### PR TITLE
feature/netlink ecmp

### PR DESCRIPTION
**Pull request by cedi**

### CODE REVIEW

File: protocols/netlink/netlink_reader.go [hunk 1]
```diff
@@ -88,66 +74,63 @@ func (nr *NetlinkReader) propagateChanges(routes []netlink.Route) {
 
 // Add given paths to clients
 func (nr *NetlinkReader) addPathsToClients(routes []netlink.Route) {
+	// If there where no routes yet, just skip this funktion. There's nothing to delete
+	if len(routes) == 0 {
+		nr.mu.RUnlock()
```
Comment (czerwonk) [inline protocols/netlink/netlink_reader.go]: unlock without prior lock => runtime error

Comment (cedi) [reply protocols/netlink/netlink_reader.go]: great finding! Thanks!

File: protocols/netlink/netlink_reader.go [hunk 2]
```diff
@@ -88,66 +74,63 @@ func (nr *NetlinkReader) propagateChanges(routes []netlink.Route) {
 
 // Add given paths to clients
 func (nr *NetlinkReader) addPathsToClients(routes []netlink.Route) {
+	// If there where no routes yet, just skip this funktion. There's nothing to delete
```
Comment (czerwonk) [inline protocols/netlink/netlink_reader.go]: copy paste from remove...? should be "There's nothing to commit"

File: protocols/netlink/netlink_reader.go [hunk 3]
```diff
@@ -88,66 +74,63 @@ func (nr *NetlinkReader) propagateChanges(routes []netlink.Route) {
...
-}
-
 // Remove given paths from clients
 func (nr *NetlinkReader) removePathsFromClients(routes []netlink.Route) {
 	nr.mu.RLock()
 
-	// get the number of routes
-	routeLength := len(nr.routes)
-
 	// If there where no routes yet, just skip this funktion. There's nothing to delete
```
Comment (czerwonk) [inline protocols/netlink/netlink_reader.go]: typo: i think you meant function

**Pull request merged**